iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -98,9 +98,13 @@ void PortManager::doDiscovery()
 void PortManager::handleSenderPorts()
 {
     // get the changes of sender port offer state
-    for (auto senderPortData : m_portPool->senderPortDataList())
+
+    auto& senderList = m_portPool->senderPortDataList();
+    for (auto iter = senderList.begin(); iter != senderList.cend();)
     {
-        SenderPortType senderPort(senderPortData);
+        auto checkIter = iter;
+        ++iter;
+        SenderPortType senderPort(checkIter.operator->());
```

Comment:
Could you please write `(*checkIter)` instead of using the complicated `operator->()` method. I think it was not intended to be used like this.